### ISSUE TITLE
Cannot connect with WordPress Android App

### ISSUE THREAD

**Issue by FatherMcGruder** [YunoHost-Apps/wordpress_ynh#36]
After entering site address, username, and passwork, in the Android app, I receive the helpful error: "Error while adding site. Error code: GENERIC_ERROR". Not sure what the source of the error could be, but I wonder if there is a default setting in the server configuration if that might prohibit Wordpress's Android app from connecting.

**Comment by FatherMcGruder** [comment]
Here is the log from the Wordpress app. It seems like there is an issue with XML-RPC.
```
WordPress - 8.7 - Version code: 485
Android device name: Google Pixel

01 - [Nov-07 19:26 UTILS] WordPress.onCreate
02 - [Nov-07 19:26 API] PostStore onRegister
03 - [Nov-07 19:26 UTILS] App comes from background
04 - [Nov-07 19:26 UTILS] Deferred Initialisation
05 - [Nov-07 19:26 STATS] Number of old stats deleted : 0
06 - [Nov-07 19:26 UTILS] Connection status changed, isConnected=true
07 - [Nov-07 19:26 NOTIFS] Sending GCM token to our remote services: [REDACTED]
08 - [Nov-07 19:26 NUX] Google API client connected
09 - [Nov-07 19:26 NUX] SmartLock: Unsuccessful credential request.
10 - [Nov-07 19:26 API] Dispatching action: SiteAction-FETCH_WPCOM_SITE_BY_URL
11 - [Nov-07 19:26 API] Volley error on https://public-api.wordpress.com/rest/v1.1/sites/[REDACTED]%2Fblog/?locale=en_US - exception: null
12 - [Nov-07 19:26 API] StackTrace: com.android.volley.ServerError
    at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:179)
    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:114)

13 - [Nov-07 19:26 API] Dispatching action: SiteAction-FETCHED_WPCOM_SITE_BY_URL
14 - [Nov-07 19:26 API] Dispatching action: AuthenticationAction-DISCOVER_ENDPOINT
15 - [Nov-07 19:26 NUX] Calling system.listMethods on the following URLs: [https://[REDACTED]/blog/xmlrpc.php, http://[REDACTED]/blog/xmlrpc.php, https://[REDACTED]/blog, http://[REDACTED]/blog, [REDACTED]/blog]
16 - [Nov-07 19:26 NUX] Trying system.listMethods on the following URL: https://[REDACTED]/blog/xmlrpc.php
17 - [Nov-07 19:26 NUX] system.listMethods replied with XML-RPC objects for https://[REDACTED]/blog/xmlrpc.php
18 - [Nov-07 19:26 NUX] Validating the XML-RPC response...
19 - [Nov-07 19:26 NUX] Validation ended with success! Endpoint found!
20 - [Nov-07 19:26 API] Dispatching action: AuthenticationAction-DISCOVERY_RESULT
21 - [Nov-07 19:26 NUX] Discovery succeeded, endpoint: https://[REDACTED]/blog/xmlrpc.php
22 - [Nov-07 19:27 UTILS] App goes to background
23 - [Nov-07 19:27 UTILS] App comes from background
24 - [Nov-07 19:27 NUX] Google API client connected
25 - [Nov-07 19:27 API] Dispatching action: SiteAction-FETCH_SITES_XML_RPC
26 - [Nov-07 19:27 API] Volley error on https://[REDACTED]/blog/xmlrpc.php - exception: org.wordpress.android.fluxc.network.xmlrpc.XMLRPCFault: 
27 - [Nov-07 19:27 API] StackTrace: com.android.volley.VolleyError: org.wordpress.android.fluxc.network.xmlrpc.XMLRPCFault: 
    at org.wordpress.android.fluxc.network.xmlrpc.XMLRPCRequest.parseNetworkResponse(XMLRPCRequest.java:64)
    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:125)
Caused by: org.wordpress.android.fluxc.network.xmlrpc.XMLRPCFault: 
    at org.wordpress.android.fluxc.network.xmlrpc.XMLSerializerUtils.deserialize(XMLSerializerUtils.java:84)
    at org.wordpress.android.fluxc.network.xmlrpc.XMLRPCRequest.parseNetworkResponse(XMLRPCRequest.java:61)
    ... 1 more

28 - [Nov-07 19:27 API] Dispatching action: SiteAction-FETCHED_SITES_XML_RPC
29 - [Nov-07 19:27 API] Login with username/pass onSiteChanged has error: GENERIC_ERROR - Error while adding site. Error code: GENERIC_ERROR
```

**Comment by sullivank16** [comment]
Was this ever resolved? I'm encountering the same issue with my website on my Moto x4.

**Comment by sullivank16** [comment]
Update: Akismet plugin was causing the issue. Disabled plugin and was able to log in.

**Comment by FatherMcGruder** [comment]
I never had Akismet enabled. I gave it another try today after I saw your message and it did not work at first. Then, on a whim, I went to the update page on my Yunohost admin site and updated everything. Suddenly, I was able to connect with the WordPress app. So, hurray!

On January 31, 2018 7:49:02 AM EST, Kelsea Sullivan <notifications@github.com> wrote:
>Update: Akismet plugin was causing the issue. Disabled plugin and was
>able to log in.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/YunoHost-Apps/wordpress_ynh/issues/36#issuecomment-361922382

-- 
Sent from my Android device with K-9 Mail. GnuPG encryption available upon request.
